docs/glossary.md

```diff
@@ -0,0 +1,62 @@
+# OLM Glossary
+
+A list of OLM terms, with definitions and common acronyms.
+
+## CustomResourceDefinitions
+
+### CatalogSources
+
+**Definition**: The CatalogSource is a repository of [CSVs](#ClusterServiceVersion), [CRDs](#CustomResourceDefinitions), and packages that define an application. 
+**Acronyms**: CatSrc(s)
+
+### ClusterServiceVersion
+
+**Definition**: The ClusterServiceVersion is a required resource for an Operator which includes application metadata such as name, description, version, repository link, labels, icon, etc. It also includes a list of corresponding/required resources and their statuses, install strategy of the Operator as [deployments](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/), supported install modes, etc.
+**Acronyms**: CSV(s)
+
+### InstallPlans
+
+**Definition**: The InstallPlan lists out all resources to be created in order to automatically install or upgrade a [CSV](#ClusterServiceVersion). 
+**Acronyms**: IP(s)
+
+### OperatorGroups
+
+**Definition**: The operator group selects a set of target namespaces in which to generate required RBAC access for its member Operators.
+**Acronyms**: OG(s)
+
+### OperatorSources
+
+**Definition**: The operator source defines the external data store used to store Operator bundles. It contains information about the catalog of operators including its name, type, namespace, and display name. 
+**Acronyms**: OpSrc(s)
+
+### Subscriptions
+
+**Definition**: The subscription defines the channel and the source of Operator updates and is used to keep the CSV updated by tracking a channel in a package. 
+**Acronyms**: Subs(s)
+
+## OLM Concepts
+
+### Bundle
+
+**Definition**: A collection of Operator manifests and metadata which together form a unique version of an Operator that can be installed onto the cluster. 
+**Acronyms**: Package
+
+### Catalog Image
+
+**Definition**: A catalog image is an image that contains Operator bundles that are stored in a database. These are then exposed over gRPC to the OLM Operator to allow users to query and install new Operators onto the cluster. 
+**Acronyms**: OPM Index
+
+### Index
+
+**Definition**: The Index refers to a database that contains information about Operator bundles including CSVs, CRDs, etc of all versions. This index can host a history of used operators on a cluster and be maintained by adding or removing operators.
```

Comment:
This makes a lot of sense to me, but our `opm` definitions are backwards...index refers to the "index image" and registry refers to the database only. (i.e `opm index add` adds to an image, not a db)

Can we massage these definitions to make that work?

Reply:
I am adding them as a parenthesis. See if that works.

Reply:
I think we want to be a little more specific. The `Index` as we have been talking about it is the ephemeral instance of the Image that encapsulates the database at a specific point in time. I.e. I would say something like:

"The Certified Operators Index doesn't have the latest version of etcd published yet", in the same way that we talk about the certified-operators app-registry namespace.

In that way, I think the Index is the actual *image* that happens to (as an implementation detail) contain a database and hosts an API to make queries about what the database contains. So it's not just the database, it's the image that is an amalgam of all of those concepts together.